### DISCUSSION TITLE
tutorials/intro-typescript/13-mutations

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#105] [course feedback]
# tutorials/intro-typescript/13-mutations

https://www.apollographql.com/tutorials/intro-typescript/13-mutations

**Comment by MarvinXu** [comment]
In the last practice, why is 'prefix the type name with mutation action' not correct?

**Reply by lizhennessy** [reply]
Hi there @MarvinXu , great callout; I can see that this answer is confusing as we use the mutation action in the name of our own input! I will make the update so this doesn't cause confusion for anyone else. :)

**Comment by MarvinXu** [comment]
```gql
input AddItemsToPlaylistInput {
  "The ID of the playlist."
  playlistId: ID!
  "A comma-separated list of Spotify URIs to add."
  uris: [String!]!
}
```
The annotation for `uris` is inappropriate since it is an array

**Reply by lizhennessy** [reply]
Hi @MarvinXu ! In the case of Schema Definition Language (SDL) syntax, we actually do refer to the brackets (`[]`) as the `List` type. It's a wrapping type in GraphQL, so it will contain a nested type to indicate what it is a _list of_, such as `[String]`. I see where you're coming from in terms of how the input is created in JSON, however! :) We were referencing [the actual Spotify API documentation](https://developer.spotify.com/documentation/web-api/reference/add-tracks-to-playlist) if you'd like to see where some of these descriptions and fields come from.

**Reply by MarvinXu** [reply]
Sorry I haven't made myself clear. What I mean is, "a comma-separated list" and "an array" are two different things. Should we change the description here to "An array of Spotify URIs to add"?

**Reply by mabuyo** [reply]
Sorry for the confusion here. Yes, we should update the description to simply read "A list of Spotify URIs to add." for clarity & correctness. Will add this on our list to fix, thanks!

**Comment by MarvinXu** [comment]
```ts
export type SnapshotOrError = {
  snapshot_id?: string;
  error?: string;
};

```
`error` has no use here. Based on the api, error returns with http code `404`, so it will throw an error on the `post` request.
```ts
const response = await dataSources.spotifyAPI.addItemsToPlaylist(input);
// you'll never get `response.error`
```
